<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1200</width>
    <height>800</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Vision Tester - YOLO Object Detection</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QHBoxLayout" name="horizontalLayout">
    <item>
     <widget class="QWidget" name="wgt_left_panel" native="true">
      <property name="maximumWidth">
       <number>400</number>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout">
       <item>
        <widget class="QGroupBox" name="grp_model">
         <property name="title">
          <string>Model Configuration</string>
         </property>
         <layout class="QVBoxLayout" name="verticalLayout_2">
          <item>
           <widget class="QLabel" name="lbl_framework_status">
            <property name="text">
             <string>Framework Status: Checking...</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLabel" name="lbl_model_select">
            <property name="text">
             <string>Select Model:</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QComboBox" name="cmb_model"/>
          </item>
          <item>
           <widget class="QTextEdit" name="txt_model_info">
            <property name="maximumHeight">
             <number>100</number>
            </property>
            <property name="readOnly">
             <bool>true</bool>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="btn_load_model">
            <property name="text">
             <string>Load Model</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QGroupBox" name="grp_settings">
         <property name="title">
          <string>Detection Settings</string>
         </property>
         <layout class="QVBoxLayout" name="verticalLayout_3">
          <item>
           <widget class="QLabel" name="lbl_confidence">
            <property name="text">
             <string>Confidence Threshold:</string>
            </property>
           </widget>
          </item>
          <item>
           <layout class="QHBoxLayout" name="horizontalLayout_2">
            <item>
             <widget class="QSlider" name="sld_confidence">
              <property name="minimum">
               <number>1</number>
              </property>
              <property name="maximum">
               <number>100</number>
              </property>
              <property name="value">
               <number>25</number>
              </property>
              <property name="orientation">
               <enum>Qt::Horizontal</enum>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QLabel" name="lbl_confidence_value">
              <property name="text">
               <string>0.25</string>
              </property>
              <property name="minimumWidth">
               <number>40</number>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item>
           <widget class="QCheckBox" name="chk_save_txt">
            <property name="text">
             <string>Save detection results as text files</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QGroupBox" name="grp_input_output">
         <property name="title">
          <string>Input/Output</string>
         </property>
         <layout class="QVBoxLayout" name="verticalLayout_4">
          <item>
           <layout class="QHBoxLayout" name="horizontalLayout_3">
            <item>
             <widget class="QPushButton" name="btn_select_image">
              <property name="text">
               <string>Select Image</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QLabel" name="lbl_image_path">
              <property name="text">
               <string>No image selected</string>
              </property>
              <property name="wordWrap">
               <bool>true</bool>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item>
           <layout class="QHBoxLayout" name="horizontalLayout_4">
            <item>
             <widget class="QPushButton" name="btn_select_input_folder">
              <property name="text">
               <string>Select Input Folder</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QLabel" name="lbl_input_folder">
              <property name="text">
               <string>No folder selected</string>
              </property>
              <property name="wordWrap">
               <bool>true</bool>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item>
           <layout class="QHBoxLayout" name="horizontalLayout_5">
            <item>
             <widget class="QPushButton" name="btn_select_output_folder">
              <property name="text">
               <string>Select Output Folder</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QLabel" name="lbl_output_folder">
              <property name="text">
               <string>No output folder selected</string>
              </property>
              <property name="wordWrap">
               <bool>true</bool>
              </property>
             </widget>
            </item>
           </layout>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QGroupBox" name="grp_actions">
         <property name="title">
          <string>Actions</string>
         </property>
         <layout class="QVBoxLayout" name="verticalLayout_5">
          <item>
           <widget class="QPushButton" name="btn_detect_single">
            <property name="text">
             <string>Detect Single Image</string>
            </property>
            <property name="enabled">
             <bool>false</bool>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="btn_detect_batch">
            <property name="text">
             <string>Detect Batch</string>
            </property>
            <property name="enabled">
             <bool>false</bool>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="btn_help">
            <property name="text">
             <string>ðŸ“– Help</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QProgressBar" name="prg_progress">
         <property name="value">
          <number>0</number>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="lbl_status_log">
         <property name="text">
          <string>Status Log:</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QTextEdit" name="txt_log">
         <property name="maximumHeight">
          <number>150</number>
         </property>
         <property name="readOnly">
          <bool>true</bool>
         </property>
        </widget>
       </item>
       <item>
        <spacer name="verticalSpacer">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>40</height>
          </size>
         </property>
        </spacer>
       </item>
      </layout>
     </widget>
    </item>
    <item>
     <widget class="QTabWidget" name="tab_display">
      <property name="currentIndex">
       <number>0</number>
      </property>
      <widget class="QWidget" name="tab_input">
       <attribute name="title">
        <string>Input Image</string>
       </attribute>
       <layout class="QVBoxLayout" name="verticalLayout_6">
        <item>
         <widget class="QLabel" name="lbl_input_image">
          <property name="text">
           <string>No image loaded</string>
          </property>
          <property name="alignment">
           <set>Qt::AlignCenter</set>
          </property>
          <property name="minimumSize">
           <size>
            <width>600</width>
            <height>400</height>
           </size>
          </property>
          <property name="styleSheet">
           <string>border: 2px solid #ccc;</string>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="tab_output">
       <attribute name="title">
        <string>Output Image</string>
       </attribute>
       <layout class="QVBoxLayout" name="verticalLayout_7">
        <item>
         <widget class="QLabel" name="lbl_output_image">
          <property name="text">
           <string>No results yet</string>
          </property>
          <property name="alignment">
           <set>Qt::AlignCenter</set>
          </property>
          <property name="minimumSize">
           <size>
            <width>600</width>
            <height>400</height>
           </size>
          </property>
          <property name="styleSheet">
           <string>border: 2px solid #ccc;</string>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1200</width>
     <height>21</height>
    </rect>
   </property>
   <widget class="QMenu" name="menu_file">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="action_exit"/>
   </widget>
   <widget class="QMenu" name="menu_help">
    <property name="title">
     <string>Help</string>
    </property>
    <addaction name="action_about"/>
   </widget>
   <addaction name="menu_file"/>
   <addaction name="menu_help"/>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <action name="action_exit">
   <property name="text">
    <string>Exit</string>
   </property>
  </action>
  <action name="action_about">
   <property name="text">
    <string>About</string>
   </property>
  </action>
 </widget>
 <resources/>
 <connections/>
</ui>
